contribution-and-developer-guidelines.md

```diff
@@ -0,0 +1,102 @@
+# CoatRack Contribution and Developer Guidelines
+
+
+
+## Introduction
+
+#### Possibilities to Contribute
+
+Hello, dear prospective contributor. You can support CoatRack in many ways:
+
+- Report bugs using GitHub's issue tracker.
+- Discuss the current state of the code in open Pull Requests.
+- Submit a bugfix.
+- Propose new features and improvement ideas.
+- Improve the documentation.
+- Clean up code.
+- Add tests.
+- Help us to fulfill the [FIWARE requirements](https://fiware-requirements.readthedocs.io/en/latest/).
+
+
+
+Note: If you want to contribute to CoatRack you need to sign the corresponding Contributors License Agreement (CLA) previously. Signing instructions can be found in [signing-tutorial.md](https://github.com/coatrack/signing-tutorial.md). 
+
+
+
+## Conventions
+
+#### Applying Best Practices
+
+The establishment of best practices in CoatRack and compliance of these is important to us. We put great effort to our review process to steadily improve the quality of our code base and we expect the same willingness from you. Here are a few practices that should always be considered: 
+
+* Code is written for readers and should be quickly understood by them. Use descriptive names, a clear architecture, short functions and small, cohesive classes. Add Java Docs if necessary.
+* Use a consistent coding style that goes along with the already existing code. Please do not change the formatting of exiting code that is not relevant to the issue you are working on.
+* Spaces should be used for indentation, not tabs.
+* Newly added code should be delivered with automated tests, for example unit tests or integration tests. Ensure to have a high code coverage.
+
+
+
+#### Workflow
+
+We use GitHub to host code, track issues and feature requests, as well as accept pull requests. We use the following workflow:
+
+1. Fork the repo. Of course, this step must be performed only once. All others are run again for each issue.
+
+2. You can either work on an existing issue or you have to create a descriptive issue before you can make code changes. 
+
+3. Create a branch from `master` for the issue you are working on. 
+
+   * Use this naming scheme: 
+     * `<commit-type>/#<issue-number>-<short-description>`. 
+     * Commits types are 'enhancement', 'feature' and 'bugfix'.
+
+   * Example: 
+     * `enhancement/#122-update-of-pom.xml-dependencies`
+
+4. If you intend to do fundamental changes, e.g. changing APIs, please contact us beforehand. This is to avoid breaking existing functionality.
+
+5. Before creating a pull request, ensure that
+   1. all tests are are passing,
+   2. the comment of the pull request contains at least this line: 
+      * `closes #<issue-number>` .
+
+6. At least two active maintainers of the projects need to review your code before it is allowed to be merged.
+
+
+
+#### How to Make a Bug Report
```

Comment:
Here should we have an example of a Bug report that follow the approach described.
I think it might be easier for the contributor to understand

Reply:
Create follow up issue about creating a issue report template

Reply:
Done, see coatrack#165.